addons/mail/controllers/discuss/public_page.py

```diff
@@ -89,23 +98,28 @@ def _response_discuss_channel_invitation(self, store, channel):
         group_public_id = channel.group_public_id or channel.parent_channel_id.sudo().group_public_id
         if group_public_id and group_public_id not in request.env.user.groups_id:
             raise request.not_found()
-        guest_already_known = channel.env["mail.guest"]._get_guest_from_context()
         with replace_exceptions(UserError, by=NotFound()):
             # sudo: mail.guest - creating a guest and its member inside a channel of which they have the token
             __, guest = channel.sudo()._find_or_create_persona_for_channel(
                 guest_name=_("Guest"),
                 country_code=request.geoip.country_code,
                 timezone=request.env["mail.guest"]._get_timezone_from_request(request),
+                join_channel=False,
             )
-        if guest and not guest_already_known:
+        # sudo: mail.guest - checking if the guest is already a member of the channel
+        guest_already_member = channel in guest.sudo().channel_ids
```

Comment:
Why do we need to `sudo()` to access our owns channels? It was not required before, and I would expect a guest to be able to access its own channels?